### ISSUE TITLE
Problem converting an existing app to nx - Uncaught TypeError - barrel problem?

### ISSUE THREAD

**Issue by maku** [nrwl/nx#214]
I have an existing app which I try to convert to nx. 

A certain part of the existing app (a UI module) should be consumable for other apps. So I created a "nx" lib (with an angular module). Because of nx is based on "barrels" I had to refactor the lib and the consuming app to be nx complient (export anything from the lib in one index.ts so that the app is able to consume it)

But now I get an error at runtime:

```
mng-model-field-boolean.component.ts:9 Uncaught TypeError: Object prototype may only be an Object or null: undefined
    at setPrototypeOf (<anonymous>)
    at __extends (mng-model-field-boolean.component.ts:9)
    at eval (mng-model-field-boolean.component.ts:20)
    at eval (mng-model-field-boolean.component.ts:20)
    at Object.../../../../../libs/ui-model/src/ui/field/mng-model-field-boolean.component.ts (main.bundle.js:925)
    at __webpack_require__ (inline.bundle.js:55)
    at eval (index.ts:1)
    at Object.../../../../../libs/ui-model/src/ui/field/index.ts (main.bundle.js:917)
    at __webpack_require__ (inline.bundle.js:55)
    at eval (index.ts:3)
```

Is this a problem with "barrels"?

I had already problem with barrels in the past with some strange errors. And some investigation also lead to this reason...

**Comment by vsavkin** [comment]
Nx builds libs as part of an app build, so there should be no barrel-related issues. If you think it is one, could you provide a repro?

**Comment by maku** [comment]
@vsavkin The error is indeed a barrel problem. 

In a naive approach I refactored all imports in a way that the "path" definition from tsconfig.json was used 

e.g.
```
import { MngModelFieldDefaultComponent } from '@offert/ui-model';
vs.
import { MngModelFieldDefaultComponent } from './mng-model-field-default.component';
```
BTW, the first version of the import is also the recommendation of my IDE (Jetbrains IntelliJ).

But the first version of the import leads to a "WARNING in Circular dependency detected" from angular cli and to the runtime error described in my first post.
 
After changing the import to the second version I was able to get the warning / runtime error away.

But I really don't get it why this is able to influence the runtime behaviour.
